src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-user-form/rgw-user-form.component.ts

```diff
@@ -74,7 +74,7 @@ export class RgwUserFormComponent extends CdForm implements OnInit {
       // General
       user_id: [
         null,
-        [Validators.required, Validators.pattern(/^[a-zA-Z0-9!@#%^&*()_-]+$/)],
+        [Validators.required, Validators.pattern(/^[a-zA-Z0-9!@$#%^&*()_-]/)],
```

Comment:
the goal here would be to not just fix the validation to allow `$` here. Rgw user has this concept of [multi-tenancy](https://docs.ceph.com/en/quincy/radosgw/multitenancy/#rgw-multi-tenancy) where multiple users can be grouped to a tenant which is done by naming the user as `<tenant>$<user>`.

If you check your screen recording, the `$` symbol that you've put in the user id, unintentionally created a tenancy for that user called `test` which is evident from the `tenant` being populated in the column.

One thing to check here are
1. can we pass something like `nizam$udeen` and expect the api to recognize it just as user id rather than `tenant$user_id`. If it doesn't can we make our api do that?